### DISCUSSION TITLE
Filter LINES in all LWPOLYLINE in DXF doc using python ezdxf

### DISCUSSION THREAD

**Discussion by Lukideluki** [mozman/ezdxf#603] [General]
Filter LINES in all LWPOLYLINE in DXF doc using python ezdxf
if someone can help me by getting the LINES coordinates from a LWPOLYLINE in a dxf doc i would appreciate.I have the following code:

```
import sys
import ezdxf
Coordenadas = str()
Coordenadas_texto = str()

try:
    doc = ezdxf.readfile(str(arq)+".dxf")
except IOError:
    print('Erro de arquivo. Certifique-se que é um arquivo .dxf')
    sys.exit(1)
except ezdxf.DXFStructureError:
    print('Arquivo invalido ou corrompoido.')
    sys.exit(2)       

msp = doc.modelspace()
            
for insert in msp.query('INSERT'):
    block = doc.blocks[insert.dxf.name]
    for e in block:
        if e.dxftype() == 'LINE':
        
            Coordenadas = Coordenadas + str(e.dxf.start) + "\n"              
            Coordenadas = Coordenadas + str(e.dxf.end) + "\n"
```
the "for" above get the block "INSERT" and split them to i can get only LINES. I tried do the same thing for LWPOLYLINE but din't worked.

the code bellow get all coordinates but i need to filter and get coordinates line by line

```
    for flag_ref in msp.query('LWPOLYLINE'):
    for entity in flag_ref:

            print(entity)
```
the code bellow stoped worked suddenly, think the librarie ezdxf changed. getting me error

_'LWPolyline' object has no attribute 'virtual_entities'_

```
for flag_ref in msp.query('LWPOLYLINE'):
    for entity in flag_ref.virtual_entities():
        if entity.dxftype() == 'LINE':

            Coordenadas = Coordenadas + str(entity.dxf.start)+ "\n"
            Coordenadas = Coordenadas + str(entity.dxf.end)+ "\n"
```

**Comment by Lukideluki** [comment]
Solved.

The last code stil working. But i am using anaconda spyder and if you install all packages for ezdxf from conda site it's subscribe this function somehow

only use `conda install -c conda-forge ezdxf`

**Comment by mozman** [comment]
The packages from `conda-forge` are not maintained by myself. The packages from [PyPI](https://pypi.org/project/ezdxf/) are always up to date, at the time of writing this 0.17.2 on `PyPI` vs. 0.17 on `conda-forge` and since 0.17.2 there are much more platforms supported by binary wheels on `PyPI` than on `conda-forge`:

### Version 0.17.2 - 2022-01-06: extended binary wheels support
- all binary wheels for Python 3.7, 3.8, 3.9 and 3.10
- `win_amd64` for Windows
- `manylinux2010_x86_64` for Python < 3.10 and `manylinux2014_x86_64` 
  for Python >= 3.10 
- `musllinux_2010_x86_64` for Python < 3.10 and `musllinux_2014_x86_64` 
  for Python >= 3.10
- `manylinux_2014_aarch64` for ARM64 based Linux
- `musllinux_2014_aarch64` for ARM64 based Linux
- `macosx_11_0_arm64` for Apple silicon
- `macosx_10_9_universal2` for Apple silicon & x86
